### DISCUSSION TITLE
Component init

### DISCUSSION THREAD

**Discussion by derynia** [arkivanov/Decompose#810] [Q&A]
Hello.
In Android viewmodel and Fragments i have onViewCreated so when I navigate to fragment it calls onViewCreated and I can make some initializations inside viewmodel's init {}
But in Decompose init of a component is called only on first navigation to the corresponding screen.
After that I can only use doOnStart() but it's not the same.
Is there a way detect that I just navigated to the component?

**Comment by arkivanov** [comment]
Currently. there is no specific callback for this case. If `doOnStart` is really not something that you could use, then I suggest to trigger the component manually from the parent component. Either call a component's method directly, or pass `Channel<Unit>` via the component's constructor and send the event when needed.

**Comment by moffpage** [comment]
Wdym I didn't get you. Init should work basically the same way within ViewModels as with Components? onViewCreated will be invoked because the corresponding fragment's view will be destroyed, but not the view model (or component in this case)
